### PR TITLE
added new workflow

### PR DESCRIPTION
**Pull request by kristinaNikolaevaa**

### CODE REVIEW

File: .github/workflows/deploy.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        out = docker_client.pull(f"{IMAGE_NAME}:{github_sha}")
+        if "error" in out:
+            raise RuntimeError(
+                f"Pull {IMAGE_NAME}:{github_sha} finished with error: {out}")
+
+        docker_client.tag(f"{IMAGE_NAME}:{github_sha}", f"{IMAGE_NAME}:{tag}")
+        out = docker_client.push(f"{IMAGE_NAME}:{tag}")
+        if "error" in out:
+            raise RuntimeError(
+                f"Push {IMAGE_NAME}:{tag} finished with error: {out}")
```
Comment (s-medvedev) [inline .github/workflows/deploy.py:101]: Please print the tag for the published image in success. It is useful.

File: .github/workflows/deploy.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    click.echo(f"Proxy branch: {proxy_branch}")
+
+    runs_before = github.get_proxy_runs_list(proxy_branch)
+
+    github.run_proxy_dispatches(proxy_branch, github_ref, github_sha, full_test_suite)
+    wait_condition(lambda: len(github.get_proxy_runs_list(proxy_branch)) > len(runs_before))
+
+    runs_after = github.get_proxy_runs_list(proxy_branch)
+    proxy_run_id = list(set(runs_after) - set(runs_before))[0]
+    click.echo(f"Proxy run id: {proxy_run_id}")
```
Comment (s-medvedev) [inline .github/workflows/deploy.py:177]: Please print a link to the proxy test run before waiting. It is helpful for debugging.

**Review by s-medvedev — COMMENTED** [review]
Looks great. There are a few comments to make CI more friendly. You can make it in the next PR.
Only remove `__pychace__` files from PR.

File: .github/workflows/pipeline.yml [hunk 3]
```diff
@@ -1,11 +1,92 @@
...
+        if: |
+          failure() &&
+           (github.ref_name == 'develop' ||
+             github.ref_name == 'master' ||
+             steps.is_version_branch.outputs.value) ||
+             startsWith(github.ref , 'refs/tags/')
+        run: |
+          python3 ./.github/workflows/deploy.py send_notification \
+          --url=${{secrets.SLACK_EVM_CHANNEL_URL}} \
+          --build_url=${BUILD_URL}
```
[the remark this answers is not in the document]
Comment (kristinaNikolaevaa) [reply .github/workflows/pipeline.yml:92]: Created https://neonlabs.atlassian.net/browse/NDEV-870

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**